tests/pipelines/test_pipelines_automatic_speech_recognition.py

```diff
@@ -699,6 +725,38 @@ def test_simple_whisper_asr(self):
                 ],
             },
         )
+        output = speech_recognizer(filename, return_timestamps="word")
+        # fmt: off
+        self.assertEqual(
+            output,
+            {
+                "text": " Mr. Quilter is the apostle of the middle classes, and we are glad to welcome his gospel.",
+                "chunks": [{
+                    "text": " Mr. Quilter is the apostle of the middle classes, and we are glad to welcome his gospel.",
+                    "timestamp": (0.0, 5.44),
+                    "words": [
+                        {'text': ' Mr.', 'tokens': [1770, 13], 'timestamp': (0.0, 1.02)},
```

Comment:
Just to chime in, this proposed return type breaks consistency with wav2vec2.

When user asks `pipeline(.., return_timestamps="word")` we should return

`{"text": full_text, 'chunks': [{"text": word1, "timestamp":(start, stop)}, ....]`

Nesting like this is a no-no in my book (it makes using the results super confusing, and worse potentially inconsistent since the timestamps of the full chunk and of each word might not even align, which *they do not here*. Making this output impossible to use as a user since there's no way of knowing who's right.

Reply:
@amyeroberts I forgot to update this test. ;-)

@Narsil Yes, it does break consistency with what the ASR pipeline promises will happen. I've asked feedback on this before and we concluded that we want to return both the sentence-level chunks and the word-level ones. I'm happy to entertain any alternative ways you'd like to return this to the user.

Reply:
Imo let's keep consistent with `wav2vec2` (ctc in general). pipelines consistency across models is one big perk imo.

You'll still get inconsistencies between return_timestamps options if the model is bad, but iiuc on good models it should be good enough.
Lower level primitives (`generate`, `tokenizer.decode` etc..) are here exactly for full control. 

@amyeroberts

Reply:
I will make this change but I think it's a mistake because we're throwing away useful information. It's useful to know what is a sentence or sentence fragment, in addition to the timing of the words in that sentence. We're losing the sentence information here, for the sake of keeping the pipeline consistent with less-capable models. Seems like the wrong trade-off to make.